File: src/main/java/pl/exsio/querydsl/entityql/examples/jpa/entity/OrderItem.java
package pl.exsio.querydsl.entityql.examples.jpa.entity;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "ORDER_ITEMS", uniqueConstraints = {
        @UniqueConstraint(name = "unique_book_for_order",
                columnNames = {"BOOK_ID", "ITEM_ORDER_ID"})
})
public class OrderItem implements Serializable {

    @Id
    @Column(name = "ORDER_ITEM_ID")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "BOOK_ID", nullable = false)
    private Book book;

    @ManyToOne
    @JoinColumn(name = "ITEM_ORDER_ID", nullable = false)
    private Order order;

    @ManyToOne
    @JoinColumn(name = "ITEM_ORDER_ID", nullable = false, insertable = false, updatable = false, referencedColumnName = "ORDER_ID")
    private Order orderReferenced;

    @Column(name = "QTY", nullable = false)
    private Long quantity;

    public OrderItem() {
    }

    public OrderItem(Long id, Book book, Order order, Long quantity) {
        this.id = id;
        this.book = book;
        this.order = order;
        this.quantity = quantity;
    }

}

File: kotlin-maven-jpa-static/src/main/kotlin/pl/exsio/querydsl/entityql/examples/jpa/entity/OrderItem.kt
package pl.exsio.querydsl.entityql.examples.jpa.entity


import pl.exsio.querydsl.entityql.examples.jpa.entity.Book
import pl.exsio.querydsl.entityql.examples.jpa.entity.Order
import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "ORDER_ITEMS", uniqueConstraints = [
    UniqueConstraint(name = "junique_book_for_order",
            columnNames = ["BOOK_ID", "ITEM_ORDER_ID"])
])
class OrderItem() : Serializable {

    @Id
    @Column(name = "ORDER_ITEM_ID")
    @GeneratedValue
    var id: Long? = null

    @ManyToOne
    @JoinColumn(name = "BOOK_ID", nullable = false)
    var book: Book? = null

    @ManyToOne
    @JoinColumn(name = "ITEM_ORDER_ID", nullable = false)
    var order: Order? = null

    @ManyToOne
    @JoinColumn(name = "ITEM_ORDER_ID", nullable = false, insertable = false, updatable = false, referencedColumnName = "ORDER_ID")
    var orderReferenced: Order? = null

    @Column(name = "QTY", nullable = false)
    var quantity: Long? = null

    constructor(id: Long, quantity: Long) : this() {
        this.id = id
        this.quantity = quantity
    }
}
